nullaway/src/test/resources/com/uber/nullaway/testdata/CheckContractNegativeCases.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2017 Uber Technologies, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.uber.nullaway.testdata;
+
+import javax.annotation.Nullable;
+import org.jetbrains.annotations.Contract;
+
+public class CheckContractNegativeCases {
+
+  @Contract("_, !null -> !null")
+  @Nullable
+  Object foo(Object a, @Nullable Object b) {
+    if (a != null) {
+      return b;
+    }
+    return new Object();
+  }
+
+  @Contract("_, !null -> !null")
+  @Nullable
+  Object fooTwo(Object a, @Nullable Object b) {
+    if (b != null) {
+      return b;
+    }
+    return new Object();
+  }
+}
```

Comment:
Overall, I feel a few more complex test cases would be illustrative here. What happens with, e.g.

```
  @Contract("!null -> !null")
  public @Nullable T orElse(@Nullable T other) {
    return value != null ? value : other;
  }
```

(from an `Optional` implementation, `value` is a field)

Or:

```
  @Contract("_, _, !null -> !null; _, !null, _ -> !null")
  @Nullable
  public String fooX(
      @Nullable Object a, @Nullable Object b, @Nullable Object c) {
    return b != null ? b : (c != null ? c : a);
  }
```

(based on a simplification of some tricky internal code)

p.s. Although in the later case, I am not 100% we handle multi-predicate contracts yet.